fdbserver/DDRelocationQueue.actor.cpp

```diff
@@ -2030,34 +2159,175 @@ ACTOR Future<bool> rebalanceReadLoad(DDQueue* self,
 	auto& metricsList = reply.shardMetrics;
 	// NOTE: randomize is important here since we don't want to always push the same shard into the queue
 	deterministicRandom()->randomShuffle(metricsList);
-	traceEvent->detail("MinReadLoad", reply.minReadLoad).detail("MaxReadLoad", reply.maxReadLoad);
+	traceEvent->detail("MinReadLoad", reply.minReadLoad)
+	    .detail("MaxReadLoad", reply.maxReadLoad)
+	    .detail("MetricListSize", metricsList.size());
 
 	if (metricsList.empty()) {
 		traceEvent->detail("SkipReason", "NoEligibleShards");
 		return false;
 	}
-
 	auto& [shard, metrics] = metricsList[0];
 	traceEvent->detail("ShardReadBandwidth", metrics.bytesReadPerKSecond);
 	//  Verify the shard is still in ShardsAffectedByTeamFailure
 	shards = self->shardsAffectedByTeamFailure->getShardsFor(
 	    ShardsAffectedByTeamFailure::Team(sourceTeam->getServerIDs(), primary));
-	for (int i = 0; i < shards.size(); i++) {
-		if (shard == shards[i]) {
-			UID traceId = deterministicRandom()->randomUniqueID();
-			self->output.send(RelocateShard(shard, moveReason, RelocateReason::REBALANCE_READ, traceId));
-			traceEvent->detail("TraceId", traceId);
 
-			auto serverIds = sourceTeam->getServerIDs();
-			self->updateLastAsSource(serverIds);
+	RelocateReason relocateReason = selectRelocateReason(self,
+	                                                     traceEvent,
+	                                                     moveReason,
+	                                                     reply.minReadLoad,
+	                                                     reply.maxReadLoad,
+	                                                     metrics.bytesReadPerKSecond,
+	                                                     srcLoad,
+	                                                     shard);
+
+	if (relocateReason == RelocateReason::OTHER) {
+		// do nothing
+		traceEvent->detail("SkipReason", "NotMoveAuxTeam");
```

Comment:
The detail needs to be more specific.

Reply:
`		traceEvent->detail("SkipReason", "NotMoveShardWithMultipleTeams");`